src/components/Visualization/Visualization.js

```diff
@@ -80,109 +81,120 @@ export const Visualization = ({
-                                    key={header.name}
-                                    name={header.name}
-                                    onSortIconClick={({ name, direction }) =>
-                                        setSorting({
-                                            sortField: name,
-                                            sortDirection: direction,
-                                        })
-                                    }
-                                    sortDirection={
-                                        header.name === sortField
-                                            ? sortDirection
-                                            : 'default'
-                                    }
-                                    large={large}
-                                    small={small}
-                                    className={fontSizeClass}
-                                >
-                                    {header.column}
-                                </DataTableColumnHeader>
-                            ) : (
-                                <DataTableColumnHeader
-                                    fixed
-                                    top="0"
-                                    key={`undefined_${index}`} // FIXME this is due to pe not being present in headers, needs special handling
-                                    large={large}
-                                    small={small}
-                                    className={fontSizeClass}
-                                />
-                            )
-                        )}
-                    </DataTableRow>
-                </DataTableHead>
-                {/* https://jira.dhis2.org/browse/LIBS-278 */}
-                <DataTableBody loading={fetching}>
-                    {data.rows.map((row, index) => (
-                        <DataTableRow key={index}>
-                            {row.map((value, index) => (
-                                <DataTableCell
-                                    key={index}
-                                    large={large}
-                                    small={small}
-                                    className={fontSizeClass}
-                                >
-                                    {formatValue(
-                                        value,
-                                        data.headers[index]?.valueType ||
-                                            'TEXT',
-                                        {
-                                            digitGroupSeparator:
-                                                visualization.digitGroupSeparator,
-                                            skipRounding: false, // TODO should there be an option for this?
+            <div
+                className={cx(styles.fetchIndicator, {
+                    [styles.fetching]: fetching,
+                })}
+            >
```

Comment:
The large diff below is mostly caused because of me adding this wrapper and causing the rest of the code to indent more. There is only one actual change in the diff below, which I will also mark with a comment.